gleambatch.py

```diff
@@ -0,0 +1,227 @@
+            batch.config["start_date"], datetime.time(23, 59, 59)
+        ).astimezone()
+        ft.apply_to_regions(rs, before=ft_before)
+
+    # TODO: This is asking for a good refactor of the redistribution code ...
+    override = batch.config.get("region_active_estimates")
+    if override is not None:
+        log.info(f"Overriding 'ft_mean' for {len(override)} regions ...")
+        for key, est in override.items():
+            rs[key].est["ft_mean"] = est
+
+    # Main computation: propagate estimates and fix for consistency with CSSE
+    FTData.propagate_down(rs)
+
+    # Propagate estimates upwards to super-regions
+    rs.fix_min_est(
+        "est_active", keep_nones=True, minimum_from="csse_active", minimum_mult=2.0
+    )  ## TODO: param for mult
+
+    rs.check_missing_estimates("est_active")
+
+    # Store the initial estimates in batch data
+    rs.fix_min_est("est_active")
+    batch.store_region_estimates(rs, "est_active", "FT_Infected")
+    # Finally, propagate csse upwards and also store
+    for loc_key, rem_key in [
+        ("csse_deaths", "JH_Deaths"),
+        ("csse_confirmed", "JH_Confirmed"),
+        ("csse_recovered", "JH_Recovered"),
+        ("csse_active", "JH_Infected"),
+    ]:
+        # Fix one-city provinces
+        for r in rs.regions:
+            if (
+                r.parent is not None
+                and len(r.parent.sub) == 1
+                and r.est.get(loc_key) is None
+            ):
+                r.est[loc_key] = r.parent.est.get(loc_key)
+        # Propagate up
+        rs.fix_min_est(loc_key)
+        # Store in batch
+        batch.store_region_estimates(rs, loc_key, rem_key)
+
+
+def estimates_to_gleamdef(batch, rs: Regions, input_xml_path):
+    gv = GleamDef(input_xml_path)
+    gv.set_start_date(batch.config["start_date"])
+    gv.clear_seeds()
+    for comp, coef in batch.config["compartments_mult"].items():
+        gv.add_seeds(rs, est_key="est_active", compartments={comp: coef})
+
+    return gv
+
+
+def parameterize(batch, gv):
+    for mit in batch.config["mitigations"]:
+        for sce in batch.config["scenarios"]:
+            gv2 = gv.copy()
+            gv2.set_seasonality(sce["param_seasonalityAlphaMin"])
+            gv2.set_traffic_occupancy(sce["param_occupancyRate"])
+            gv2.set_beta(mit["param_beta"])
+            gv2id = "{}.574".format(random.randint(1700000000000, 1800000000000))
+            gv2.set_id(gv2id)
+            gv2.set_name(gv2.full_name(batch.name))
+            sim = Simulation(gv2, None)
+            batch.add_simulation_info(
+                sim,
+                name=sce["name"],
+                group=mit["label"],
+                color=sce.get("color"),
+                style=sce.get("style"),
+            )
+
+
+def generate(args):
+    "Run the 'generate' subcommand"
+
+    with open(args.CONFIG_YAML, "rt") as f:
+        config = yaml.load(f)
+    Path(config["output_dir"]).expanduser().mkdir(exist_ok=True, parents=True)
```

Comment:
I would prefer to stat with `os.mkdir(Path...)`, as my first reaction was "Statement has no effect", but agree that this is sexy.